src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -367,3 +368,85 @@ def _try_fetch(self) -> Iterable[MigrationStatus]:
     def _iter_schemas(self):
         for catalog in self._ws.catalogs.list():
             yield from self._ws.schemas.list(catalog_name=catalog.name)
+
+
+class TablesInMountsMigrator:
+    def __init__(
+        self,
+        tables_in_mounts: TablesInMounts,
+        ws: WorkspaceClient,
+        backend: SqlBackend,
+        table_mapping: TableMapping,
+    ):
+        self._backend = backend
+        self._ws = ws
+        self._tm = table_mapping
+        self._tim = tables_in_mounts
+
+    def create_tables_in_uc(self):
+        create_table_tasks = []
+        mapping_keys = {mapping.src_table: mapping for mapping in self._tm.load()}
+        for table in self._tim.fetch_all():
+            if self._tim.TABLE_IN_MOUNT_DB not in table.database:
+                logger.info(f"Path {table.location} hasn't been identified as a table in a mount, ignoring")
+                continue
+            if not table.location:
+                continue
+            if table.location not in mapping_keys:
```

Comment:
Isn't the `table.location` the full path of the location, while the keys in `mapping_keys` are the source table name which is the `os.path.basename(path)` in this case? Should the code extract the basename from the `table.location` here and then check if it's in `mapping_keys`? Or do I misunderstand it?

Reply:
the keys in mapping_keys are the source table name, but it's not `os.path.basename(path)`. It's actually the full table path.

Basically we're checking here if the table path exists in the mapping file. Because the mapping file maps a table path to a catalog.database.table

Reply:
But the `os.path.basename(path)` is put as the table name [here](https://github.com/databrickslabs/ucx/blob/362c5475f2b8d6f38175787d61b48d14409dbe86/src/databricks/labs/ucx/hive_metastore/locations.py#L331C1-L335C38)